### ISSUE TITLE
Pack icon.png does not display

### ISSUE THREAD

**Issue by skjbulcher** [StackStorm/st2#3128]
I'm having trouble getting my pack's [icon.png](https://drive.google.com/file/d/0B3QFylZ1CuvsYTVJR1dzODR5YjA/view?usp=sharing) to display.

Running st2 version 2.1.1

**permissions** `/opt/stackstorm/packs`
```
root@stackstorm:/opt/stackstorm/packs# ls -rlth
total 24K
drwxrwxr-x 5 root st2packs 4.0K Dec 16 21:36 packs
drwxrwxr-x 4 root st2packs 4.0K Dec 16 21:36 linux
drwxrwxr-x 5 root st2packs 4.0K Dec 16 21:36 default
drwxrwxr-x 5 root st2packs 4.0K Dec 16 21:36 chatops
drwxrwxr-x 4 root st2packs 4.0K Dec 19 19:09 core
drwxrwxr-x 4 root st2packs 4.0K Dec 19 19:29 grade_us
```

**permissions** `/opt/stackstorm/packs/grade_us`
```
root@stackstorm:/opt/stackstorm/packs/grade_us# ls -rlth
total 28K
-rwxrwxr-x 1 root st2packs   22 Dec 19 19:29 requirements.txt
-rwxrwxr-x 1 root st2packs  206 Dec 19 19:29 pack.yaml
-rwxrwxr-x 1 root st2packs 3.8K Dec 19 19:29 icon.png
-rwxrwxr-x 1 root st2packs   52 Dec 19 19:29 grade_us.yaml.example
-rwxrwxr-x 1 root st2packs   94 Dec 19 19:29 config.schema.yaml
drwxrwxr-x 3 root st2packs 4.0K Dec 19 19:29 actions
```

**output** `st2api.log` after loading URL https://<mystackstorm>.com/api/v1/packs/views/file/grade_us/icon.png?x-auth-token=TOKEN
```
root@stackstorm:/var/log/st2# tail -f -n 0 st2api.log
2016-12-19 19:45:06,990 140160140703856 INFO hooks [-] 1b150dd3-fbaf-485c-9e06-afae8e3e5433 -  GET /v1/packs/views/file/grade_us/icon.png with filters={} (remote_addr='127.0.0.1',method='GET',filters={},request_id='1b150dd3-fbaf-485c-9e06-afae8e3e5433',path='/v1/packs/views/file/grade_us/icon.png')
2016-12-19 19:45:06,991 140160140703856 AUDIT auth [-] Token provided in query parameters
2016-12-19 19:45:06,993 140160140703856 AUDIT auth [-] Token with id "585820f802ebd5032fdb79e0" is validated.
2016-12-19 19:45:06,997 140160140703856 INFO hooks [-] 1b150dd3-fbaf-485c-9e06-afae8e3e5433 - GET /v1/packs/views/file/grade_us/icon.png (remote_addr='127.0.0.1',method='GET',status_code='404 Not Found',request_id='1b150dd3-fbaf-485c-9e06-afae8e3e5433',path='/v1/packs/views/file/grade_us/icon.png')
```

Core pack icons are accessible, as are at least some of the files in their directories, whereas in my *grade_us* pack, no files are accessible via the URL.

**Comment by LindsayHill** [comment]
@skjbulcher I just came across something that reminded me of a possible solution: is your `pack.yaml` file up to date? In particular, check the `name` and `ref` entries in there. Are they correct? Or do they have the incorrect name?

**Comment by Mierdin** [comment]
@skjbulcher I was able to retrieve the image for a pack I'm working on, as well as that of the core pack:

```
[vagrant@st2vagrant ~]$ curl -v -s http://127.0.0.1:9101/v1/packs/views/file/core/icon.png?x-auth-token=577f6bf18ba9421e955b0d7734e49b19 1> /dev/null
* About to connect() to 127.0.0.1 port 9101 (#0)
*   Trying 127.0.0.1... connected
* Connected to 127.0.0.1 (127.0.0.1) port 9101 (#0)
> GET /v1/packs/views/file/core/icon.png?x-auth-token=577f6bf18ba9421e955b0d7734e49b19 HTTP/1.1
> User-Agent: curl/7.19.7 (x86_64-redhat-linux-gnu) libcurl/7.19.7 NSS/3.21 Basic ECC zlib/1.2.3 libidn/1.18 libssh2/1.4.2
> Host: 127.0.0.1:9101
> Accept: */*
>
< HTTP/1.1 200 OK
< Server: gunicorn/19.6.0
< Date: Mon, 19 Dec 2016 23:28:12 GMT
< Connection: keep-alive
< Last-Modified: Thu, 15 Dec 2016 23:38:56 GMT
< ETag: 1481845136.0
< Content-Type: image/png
< Content-Length: 3936
< Access-Control-Allow-Origin: http://127.0.0.1:3000
< Access-Control-Allow-Methods: GET,POST,PUT,DELETE,OPTIONS
< Access-Control-Allow-Headers: Content-Type,Authorization,X-Auth-Token,St2-Api-Key,X-Request-ID
< Access-Control-Expose-Headers: Content-Type,X-Limit,X-Total-Count,X-Request-ID
< X-Request-ID: bcd8fbd7-8a40-4b45-b877-6852466914b0
<
{ [data not shown]
* Connection #0 to host 127.0.0.1 left intact
* Closing connection #0
[vagrant@st2vagrant ~]$ curl -v -s http://127.0.0.1:9101/v1/packs/views/file/napalm/icon.png?x-auth-token=577f6bf18ba9421e955b0d7734e49b19 1> /dev/null
* About to connect() to 127.0.0.1 port 9101 (#0)
*   Trying 127.0.0.1... connected
* Connected to 127.0.0.1 (127.0.0.1) port 9101 (#0)
> GET /v1/packs/views/file/napalm/icon.png?x-auth-token=577f6bf18ba9421e955b0d7734e49b19 HTTP/1.1
> User-Agent: curl/7.19.7 (x86_64-redhat-linux-gnu) libcurl/7.19.7 NSS/3.21 Basic ECC zlib/1.2.3 libidn/1.18 libssh2/1.4.2
> Host: 127.0.0.1:9101
> Accept: */*
>
< HTTP/1.1 200 OK
< Server: gunicorn/19.6.0
< Date: Mon, 19 Dec 2016 23:28:29 GMT
< Connection: keep-alive
< Last-Modified: Mon, 19 Dec 2016 21:04:45 GMT
< ETag: 1482181485.0
< Content-Type: image/png
< Content-Length: 38031
< Access-Control-Allow-Origin: http://127.0.0.1:3000
< Access-Control-Allow-Methods: GET,POST,PUT,DELETE,OPTIONS
< Access-Control-Allow-Headers: Content-Type,Authorization,X-Auth-Token,St2-Api-Key,X-Request-ID
< Access-Control-Expose-Headers: Content-Type,X-Limit,X-Total-Count,X-Request-ID
< X-Request-ID: 242905d8-cff9-4d01-89a3-aa75eb8c42b4
<
{ [data not shown]
* Connection #0 to host 127.0.0.1 left intact
* Closing connection #0
[vagrant@st2vagrant ~]$ ll /opt/stackstorm/packs/napalm/
total 88
drwxr-xr-x. 2 vagrant vagrant  4096 Dec 13 00:31 actions
-rwxr-xr-x. 1 vagrant vagrant   433 Dec 13 00:32 circle.yml
-rw-r--r--. 1 vagrant vagrant   702 Dec 13 00:31 config.schema.yaml
-rwxr-xr-x. 1 vagrant vagrant  5257 Dec 12 22:42 exchange-bootstrap.sh
-rw-r--r--. 1 vagrant vagrant 38031 Dec 19 21:04 icon.png
-rw-r--r--. 1 vagrant vagrant   192 Dec 13 00:31 napalm.yaml.example
-rw-r--r--. 1 vagrant vagrant   204 Dec 13 00:31 pack.yaml
-rw-r--r--. 1 vagrant vagrant  1084 Dec 13 00:31 README.md
-rw-r--r--. 1 vagrant vagrant   134 Dec 13 00:31 requirements.txt
drwxr-xr-x. 2 vagrant vagrant  4096 Dec 13 00:31 rules
drwxr-xr-x. 2 vagrant vagrant  4096 Dec 13 00:31 sensors
```

Based on what I saw in chat, I noticed you've run `st2ctl reload --register-all` at least a few times;  can you post your `pack.yaml`?

**Comment by skjbulcher** [comment]
pack.yaml
```
---
ref: grade_us
name: Grade.us Integrations
description: Integration with the Grade.us reputation management platform
version: 0.1.0
author: John Bulcher
email: jbulcher@starkartthenala.com
```

**Comment by LindsayHill** [comment]
That all looks good, unless there's some ref/name weirdness messing us up. No duplicate pack names or references anywhere?

**Comment by Kami** [comment]
@skjbulcher To make sure it's not related to the ref / name change (it could be a bug), can you please try changing the name to `grade_us`?

**Comment by Kami** [comment]
@skjbulcher The issue is indeed related to the problem I mentioned in my comment above.

I have a fix in https://github.com/StackStorm/st2/pull/3131 which still needs to be finished. This fix should be included in v2.0.2.

Thanks again for reporting this.

**Comment by skjbulcher** [comment]
Yes, that did the trick. Thanks for your help!

**Comment by Kami** [comment]
@skjbulcher Great to hear that and you are welcome.

And as mentioned above, fix for that will include be included in a future release (v2.1.2) :).

**Comment by Kami** [comment]
@skjbulcher Fix in #3131 has been merged into master so I will close this issue.

I will let you know when v2.1.2 with this fix is out and at that time you can test it and re-open the ticket in case the issue still exists or similar.
